### DISCUSSION TITLE
returning response breaks type

### DISCUSSION THREAD

**Discussion by turkerdev** [trpc/trpc#1745] [Q&A]
the prisma types work normally, but as soon as I return or try to return one of its properties the type gets break.

![image](https://user-images.githubusercontent.com/48413508/162109786-28c3850c-5018-445e-8017-85d8e94db9a4.png)

also, I don't know even know why there is `then function` type inside return type
![image](https://user-images.githubusercontent.com/48413508/162110284-37dcb334-56b3-42b9-98a4-563db4b84c49.png)

```ts
export const akaRouter = createRouter().mutation("create", {
  input: z.object({
    target: z.string(),
  }),
  output: z.object({
    short: z.string(),
  }),
  resolve: async ({ ctx, input }) => {
    const exist = await prisma.aka.findFirst(...);

    if (exist) {
      throw new TRPCError({
        code: "BAD_REQUEST",
        message: "already exists",
      });
    }

    const aka = await prisma.aka.create(...);

    return aka
  },
});

```

**Comment by jlalmes** [comment]
I'm not sure that I fully understand the issue. The code you posted above looks like it should work correctly.

It seems to me that you issue is `await prisma.aka.create(...)` is returning `any` type. This will be an issue with Prisma (not tRPC), please make sure that you have run `prisma generate` recently.

If you are still facing issues, can you please share a reproduction and I'll take a look!

PS. 
`then` is a possible property because a resolver return type is `TOutput | Promise<TOutput>`. 
You can read more about the [`PromiseLike` interface here](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_es5_d_.promiselike.html).

**Reply by turkerdev** [reply]
tried to reproduce the issue on codesandbox but I couldn't. maybe its vscode issue? idk.
the issue seems like related to resolve parameters, when I use ctx or input parameter and return it, this happens.
![image](https://user-images.githubusercontent.com/48413508/162249377-ff34be8d-9abd-42ba-b970-ee0b4a736752.png)

**Reply by LouisHaftmann** [reply]
are you using typescript's `strict` compiler option?

**Reply by turkerdev** [reply]
yes

**Reply by uncvrd** [reply]
<img width="799" alt="image" src="https://user-images.githubusercontent.com/25097099/167995577-0e756dd5-2360-41fa-96ed-81c8d7e11f1f.png">

For what it's worth, this also annoyingly happens at random times with other libraries (not just Prisma). For example, my types from the Twilio API library will sometimes return as `any` as well. Pretty much impossible to recreate in a codesandbox and pretty sure it has to do with VS Code but never had this problem until integrating TRPC (which I love besides this problem haha)

**Reply by pikaju** [reply]
I have the same issue, did anyone find a solution yet? Virtually every time I add a return statement to my queries, all types die, everything is any and I don't get autocomplete for prisma or other context stuff anymore

**Reply by uncvrd** [reply]
it's a (little) bit better in v10 but I'll still go through periods of time where ALL my typings just break at once in all my open files, once I make a change and save a file, all my typings slowly go back to normal 🤷

**Reply by talksik** [reply]
I worked around it by putting an explicit return type for each handler in each router. 

I noticed that the typescript inferrence struggles inferring types when it is trying to find out types of subsequent files. In this case, the app router has handlers that call methods in other files which have their own inferred returned type.

**Comment by QastanFIT** [comment]
Currently having the same problem. Any fixes for trpc causing everything to be "any"?
